src/lib.rs

```diff
@@ -73,6 +73,29 @@ impl<T: Clone> Drop for Pinboard<T> {
     }
 }
 
+/// An wrapper around a `Pinboard` which provides the guarantee it is never empty.
+pub struct NonEmptyPinboard<T: Clone>(Pinboard<T>);
+
+impl<T: Clone> NonEmptyPinboard<T> {
+    /// Create a new `NonEmptyPinboard` instance holding the given value.
+    pub fn new(t: T) -> NonEmptyPinboard<T> {
+        NonEmptyPinboard(Pinboard::new(t))
+    }
+
+    /// Update the value stored in the `NonEmptyPinboard`.
+    pub fn set(&self, t: T) {
```

Comment:
It's worth marking it at at least `#[inline]` so the bytecode is made available for linked crates (rather than just inlining in this crate).